pkg/config/node.go

```diff
@@ -291,7 +292,7 @@ func IsUpgradeStillRunning(kubeconfigPath string) (bool, error) {
 	return true, nil
 }
 
-func GetIngressConfig(kubeconfigPath string, filterIpType string) (ingressConfig IngressConfig, err error) {
+func GetIngressConfig(kubeconfigPath, apiVip, ingressVip string) (ingressConfig IngressConfig, err error) {
```

Comment:
I thought about it, but the problem is that I need to have *both* vips whenever I generate a single keepalived configuration stanza. Look at the following example

API VIP - 1.1.1.101
Ingress VIP - 1.1.1.102
NodeIP - 1.1.1.50

I have a node which holds VIPs, so its list of IP addresses looks like `[1.1.1.50,1.1.1.101,1.1.1.102]`. If I passed only single VIP as a filter, now what happens...

We want to generate config for API, so we pass 1.1.1.101 to filter it out. The function will take all the IPs of the node (`[1.1.1.50,1.1.1.101,1.1.1.102]`), remove API VIP from this list so we end up with `[1.1.1.50,1.1.1.102]`. So now, the function has full authority to select `1.1.1.102` as the NodeIP to use in building the peer list. This is wrong, because `1.1.1.102` is Ingress VIP. The mistake happened because the function acted locally and had no idea there are other VIPs in this environment.

Now, if the function allows to filter out two addresses, what happens is... We want to generate config for API as previously. We pass `.101` and `.102` to filter out, saying those are both VIPs and should never make it to the NodeIP. So our list of potential addresses is very simple, `[1.1.1.50]`. This way, there is no possibility that we select a VIP as an entry in the peer list.

I agree, the naming here can be somehow confusing. I wonder if maybe I should change the function signature to have `filterIPs []string` so that you could just pass a list of IPs that should be forbidden from being selected? What do you think?

The general issue is that indeed the function `GetIngressConfig` is used to generate both API VIP and Ingress VIP configs, but can't accept only single IP for reasons described above. I guess this is just a naming confusion as "ingress" in keepalived world is different than "ingress" in OpenShift world

Reply:
Oh, I didn't think about the fact that you need to filter too. Most places we just use the VIP to determine the subnet or the ip version, but in this case you do need to ignore both. Since this was nit-picking anyway I don't think we need to change anything.